virtwho/virt/esx/esx.py

```diff
@@ -326,11 +326,29 @@ def getVmUuid(self, vm):
         From: 78563412-AB90-EFCD-1234-567890ABCDEF
         To:   12345678-90AB-CDEF-1234-567890ABCDEF
         """
+        smbios_version_27 = None
         s = vm['config.uuid']
-        if 'config.version' not in vm:
-            return s
-        version = int(vm['config.version'].split('-')[1])
-        if (version >= 13):
+
+        # Try to some crazy heuristics
+        if "config.extraConfig" in vm:
+            extra_config = vm["config.extraConfig"]
+            for item in extra_config.OptionValue:
+                if item.key == "acpi.smbiosVersion2.7":
+                    if item.value == "TRUE":
+                        smbios_version_27 = True
+                    elif item.value == "FALSE":
+                        smbios_version_27 = False
+
+        self.logger.debug(f"ESX acpi.smbiosVersion2.7: {smbios_version_27}")
+
+        version = 0
+        if 'config.version' in vm:
+            version = int(vm['config.version'].split('-')[1])
```

Comment:
Are all config.version values of the same form? Can we be sure this won't result in an IndexError if we attempt to access [1] on the result of the split, if in case the format is different?